capi/build.rs

```diff
@@ -93,28 +126,29 @@ fn compile(compiler: &str, src: &Path, dst: &str, options: &[&str]) {
             .chain([src.as_os_str(), "-o".as_ref(), dst.as_os_str()]),
     )
     .unwrap_or_else(|err| panic!("failed to run `{compiler}`: {err}"));
+
+    let () = adjust_mtime(&dst).unwrap();
```

Comment:
maybe split this into a separate commit and add a comment on why we are doing this?